torch/nn/functional.py

```diff
@@ -1831,3 +1832,36 @@ def normalize(input, p=2, dim=1, eps=1e-12):
         eps (float): small value to avoid division by zero. Default: 1e-12
     """
     return input / input.norm(p, dim, True).clamp(min=eps).expand_as(input)
+
+
+def unfold(input, kernel_size, dilation=1, padding=0, stride=1):
+    r"""
+    See :class:`torch.nn.Unfold` for details
+    """
+    if input is not None and input.dim() == 4:
+        assert len(kernel_size) == 1 or len(kernel_size) == 2, 'kernel_size must be length 1 or 2 for 4D input'
```

Comment:
Let's say `kernel_size` is an int. Then `len(kernel_size)` is invalid syntax. Same for checking dilation, padding, and stride below.